dl/src/main/scala/com/intel/analytics/bigdl/dataset/text/Tokenizer.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to Intel Corporation under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * Intel Corporation licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.dataset.text
+
+import com.intel.analytics.bigdl.dataset.Transformer
+
+import scala.collection.Iterator
+
+/**
+ * Transformer that tokenizes a String into a Seq[String] by splitting on a regular expression.
+ *
+ * @param sep the delimiting regular expression to split on.
+ *            Defaults to matching all punctuation and whitespace
+ */
+class Tokenizer(sep: String) extends Transformer[String, Seq[String]] {
+  override def apply(prev: Iterator[String]): Iterator[Seq[String]] =
+    prev.map(x => x.split(sep))
+}
+
+object Tokenizer {
```

Comment:
Thanks for reminder, spark mllib's RegexTokenizer is exactly what I want, maybe I need to see how to use it and insert to the transformer pipeline